OpenRA.Game/Actor.cs

```diff
@@ -96,7 +87,14 @@ internal Actor(World world, string name, TypeDictionary initDict)
 
 				Info = world.Map.Rules.Actors[name];
 				foreach (var trait in Info.TraitsInConstructOrder())
+				{
 					AddTrait(trait.Create(init));
+
+					// Some traits rely on properties provided by IOccupySpace in their initialization,
+					// so we must ready it now, we cannot wait until all traits have finished construction.
+					if (trait is IOccupySpaceInfo)
+						OccupiesSpace = Trait<IOccupySpace>();
```

Comment:
Should be something like:
```c#
	foreach (var trait in Info.TraitsInConstructOrder())
	{
		var instance = trait.Create(init);
		AddTrait(instance);

		// Some traits rely on properties provided by IOccupySpace in their initialization,
		// so we must ready it now, we cannot wait until all traits have finished construction.
		if (instance is IOccupySpace)
		{
			if (OccupiesSpace == null)
				OccupiesSpace = instance;
			else
				throw InvalidDataException("Multiple OccupySpace traits");
		}
	}
```
The trait lookup is not needed when the trait instance is created in the same block.

This is also, currently, the only way catch multiple traits implementing `IOccupySpace`.